doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -476,14 +504,138 @@ In Ltac2, we have the following backtracking primitives, defined in the
   val plus : (unit -> 'a) -> (exn -> 'a) -> 'a
   val case : (unit -> 'a) -> ('a * (exn -> 'a)) result
 
-If one views thunks as lazy lists, then `zero` is the empty list and `plus` is
-list concatenation, while `case` is pattern-matching.
+`zero` takes an exception and raises it. It can be understood as the
+trivial exception handler which performs no handling.
+
+`plus` takes a computation `r` and an exception handler `h`, and runs
+the computation `r` under the handler `h`. If `r ()` returns a value
```

Comment:
@ppedrot This line might be problematic from your POV because it suggests that the purpose of the handler `h` is to handle exceptions arising from `r`, but it is certainly true that if `r` fails, `h` is responsible for handling the exception (i.e., the concatenation of the empty stream with `h`, which I think is an important use case.) In response to previous comments you made, I clarified the paragraph afterwards to indicate that `plus r h` itself does implicitly return a continuation if `r` succeeds, and `h` is a part of that continuation (along with the continuation implicitly returned by `r`).